dev_tools/nyaml2nxdl/nyaml2nxdl_forward_tools.py

```diff
@@ -0,0 +1,1242 @@
+        "optional",
+        "stride",
+    ]
+
+    xml_handle_comment(obj, line_annot, line_loc)
+    l_bracket = -1
+    r_bracket = -1
+    if keyword.count("(") == 1:
+        l_bracket = keyword.index("(")
+    if keyword.count(")") == 1:
+        r_bracket = keyword.index(")")
+
+    keyword_name, keyword_type = nx_name_type_resolving(keyword)
+    if not keyword_type and not keyword_name:
+        raise ValueError("Check for name or type in field.")
+    elemt_obj = ET.SubElement(obj, "field")
+
+    # type come first
+    if l_bracket == 0 and r_bracket > 0:
+        elemt_obj.set("type", keyword_type)
+        if keyword_name:
+            elemt_obj.set("name", keyword_name)
+    elif l_bracket > 0:
+        elemt_obj.set("name", keyword_name)
+        if keyword_type:
+            elemt_obj.set("type", keyword_type)
+    else:
+        elemt_obj.set("name", keyword_name)
+
+    if value:
+        rm_key_list = []
+        # In each each if clause apply xml_handle_comment(), to collect
+        # comments on that yaml line.
+        for attr, vval in value.items():
+            if "__line__" in attr:
+                continue
+            line_number = f"__line__{attr}"
+            line_loc = value[line_number]
+            if attr == "doc":
+                xml_handle_doc(
+                    elemt_obj,
+                    vval,
+                    line_number,
+                    line_loc,
+                )
+                rm_key_list.append(attr)
+                rm_key_list.append(line_number)
+            elif attr == "exists" and vval:
+                xml_handle_exists(value, elemt_obj, attr, vval)
+                rm_key_list.append(attr)
+                rm_key_list.append(line_number)
+                xml_handle_comment(obj, line_number, line_loc, elemt_obj)
+            elif attr == "unit":
+                xml_handle_units(elemt_obj, vval)
+                xml_handle_comment(obj, line_number, line_loc, elemt_obj)
+            elif attr in allowed_attr and not isinstance(vval, dict) and vval:
+                validate_field_attribute_and_value(attr, vval, allowed_attr, value)
+                elemt_obj.set(attr, check_for_mapping_char_other(vval))
+                rm_key_list.append(attr)
+                rm_key_list.append(line_number)
+                xml_handle_comment(obj, line_number, line_loc, elemt_obj)
+
+        for key in rm_key_list:
+            del value[key]
+        # Check for skipped attrinutes
+        check_for_skiped_attributes("field", value, allowed_attr, verbose)
+
+    if isinstance(value, dict) and value != {}:
+        recursive_build(elemt_obj, value, verbose)
+
+
+def xml_handle_comment(
+    obj: ET.Element,
+    line_annotation: str,
+    line_loc_no: int,
+    xml_ele: ET.Element = None,
+    is_def_cmnt: bool = False,
+):
+    """
+        Add xml comment: check for comments that has the same 'line_annotation'
+    (e.g. __line__data) and the same line_loc_no (e.g. 30). After that, i
```

Comment:
"After that, it does the following tasks:"?

Reply:
:+1: